### PR TITLE
fuzz: store the corpus

### PR DESCRIPTION
**Pull request by tommady**
fulfill #5312

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
Small tip: if you use one of the keywords that GitHub recognizes (i.e `fixes` or `closes` instead of `fulfill`) it will automatically link the issue to the PR: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

**Comment by tommady** [comment]
```
Note: You can only download artifacts in a workflow that were uploaded during the same workflow run.
```

hi @sylvestre
I am sorry the github artifacts way is not working.

specifically, the download can only be used in the same workflow as the doc said,
so maybe we should consider creating a new repo to store the corpus.

so the new plan:
1. a new repo for storing the corpus
2. github action upload or commit corpus in there
3. github action download or clone corpus from there

or I am sure the artifact has been uploaded into this repo,
but I don't have the permission to see it,
if I can we could consider using the github cli instead for the downloading.

WDYT?

**Comment by sylvestre** [comment]
i think you might want to try the cache storage? no?

**Comment by sylvestre** [comment]
new repo sounds great too :)

**Comment by tommady** [comment]
> i think you might want to try the cache storage? no?

That sounds great, let me do some study and try on this
https://github.com/actions/cache

thank you

**Comment by tommady** [comment]
Hi @sylvestre 
the cache action working!
image

with and without cache the time compression
```bash
https://github.com/uutils/coreutils/actions/runs/6468556171/job/17560815053
9m 36s
vs
https://github.com/uutils/coreutils/actions/runs/6491029799/job/17627703803?pr=5363
8m 51s
```

**Comment by sylvestre** [comment]
and it seems that it works:
`2023-10-12T03:56:14.2306567Z INFO:      854 files found in /home/runner/work/coreutils/coreutils/fuzz/corpus/fuzz_date`

**Comment by sylvestre** [comment]
Bravo :)

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -1011,5 +1011,4 @@ jobs:
         ## flags: IntegrationTests, UnitTests, ${{ steps.vars.outputs.CODECOV_FLAGS }}
         flags: ${{ steps.vars.outputs.CODECOV_FLAGS }}
         name: codecov-umbrella
-        fail_ci_if_error: false
-
+        fail_ci_if_error: false
```
Comment (sylvestre) [inline .github/workflows/CICD.yml:1014]:
```suggestion
        fail_ci_if_error: false

```

File: .github/workflows/fuzzing.yml [hunk 2]
```diff
@@ -24,41 +24,45 @@ jobs:
...
       run: cargo install cargo-fuzz
     - uses: Swatinem/rust-cache@v2
+    - name: Restore Cached Corpus
+      uses: actions/cache/restore@v3
+      with:
+        key: corpus-cache
+        path: |
+          fuzz/corpus
     - name: Run fuzz_date for XX seconds
+      continue-on-error: true
```
Comment (sylvestre) [inline .github/workflows/fuzzing.yml:34]: why this change?

Comment (sylvestre) [reply .github/workflows/fuzzing.yml:34]: and the others
continue-on-error: true

Comment (tommady) [reply .github/workflows/fuzzing.yml:34]: because the fuzz test like fuzz_date will panic on a huge number as input so I want every step in this workflow to be passed then I can make sure the cache strategy is working.

but right now I need your guidance to tell me which one needs the 
```bash
continue-on-error: true
```
which one don't need.

thank you.

Comment (sylvestre) [reply .github/workflows/fuzzing.yml:34]: we need it for expr and date. the rest should pass

**Comment by sylvestre** [comment]
Many thanks

**Comment by sylvestre** [comment]
and it works!

`INFO:       93 files found in /home/runner/work/coreutils/coreutils/fuzz/corpus/fuzz_expr`
well done
